### PR TITLE
Switch to new fold algorithm & fix lg participants tagging.

### PR DESCRIPTION
**Pull request by IsaackMwenda**
Copied from https://github.com/AfricasVoices/Project-IOM/pull/22, This updates WUSC to bring it in line with a number of improvements to IOM and new Core Data folding strategy with the following extra change.
- Move listening group participants tagging to happen after folding to fix an issue where LG participants who have engaged more than once were being tagged as false during boolean keys reconciliation before exporting analysis CSV. Code reference https://github.com/AfricasVoices/Project-WUSC-KEEP-II/commit/f84e90e6d10e5bf5e9c055853baee40d6aa9430b.

### CODE REVIEW

File: src/analysis_file.py [hunk 1]
```diff
@@ -96,55 +43,72 @@ def generate(user, data, pipeline_configuration, csv_by_message_output_path, csv
             td.append_data(analysis_dict,
                            Metadata(user, Metadata.get_call_location(), TimeUtils.utc_now_as_iso_string()))
 
+        # Hide data from participants who opted out
+        ConsentUtils.set_stopped(user, data, consent_withdrawn_key, additional_keys=export_keys)
+
+        #log.info("Tagging listening group participants")
```
Comment (as2388) [inline src/analysis_file.py]: Log or remove

File: src/analysis_file.py [hunk 2]
```diff
@@ -96,55 +43,72 @@ def generate(user, data, pipeline_configuration, csv_by_message_output_path, csv
...
+        ListeningGroups.tag_listening_groups_participants(user, data, pipeline_configuration, raw_data_dir)
```
Comment (as2388) [inline src/analysis_file.py]: If you do this before setting consent then you won't leak any information about listening groups through to the analysis file when you shouldn't, which reduces the risk of accidentally analysing listening group participants who opted out.

**Review by as2388 — APPROVED** [review]
Please confirm you tested before + after and get the same outputs before merging

File: src/analysis_file.py (hunk 1 above)
Comment (IsaackMwenda) [reply src/analysis_file.py]: thanks for catching this.

File: src/analysis_file.py (hunk 2 above)
Comment (IsaackMwenda) [reply src/analysis_file.py]: updated thanks.

**Comment by IsaackMwenda** [comment]
> Please confirm you tested before + after and get the same outputs before merging
The analysis output is similar with the expected difference in lg participation tags

**Pull request merged**